pyrenew_hew/pyrenew_hew_data.py

```diff
@@ -118,12 +132,179 @@ def n_days_post_init(self):
+                            - site_indices.select(pl.col("site_pop"))
+                            .sum()
+                            .to_numpy()
+                            .flatten()[0]
+                        ],
+                    }
+                )
+            else:
+                aux_subpop = pl.DataFrame()
+            site_subpop_spine = (
+                pl.concat([aux_subpop, site_indices], how="vertical_relaxed")
+                .with_columns(
+                    [
+                        pl.col("site_index").cum_count().alias("subpop_index"),
+                        pl.when(pl.col("site").is_not_null())
+                        .then(
+                            pl.col("site").map_elements(
+                                lambda x: f"Site: {x}", return_dtype=str
+                            )
+                        )
+                        .otherwise(pl.lit("remainder of population"))
+                        .alias("subpop_name"),
+                    ]
+                )
+                .rename({"site_pop": "subpop_pop"})
+            )
+        else:
+            site_subpop_spine = pl.DataFrame(
+                {
+                    "site_index": [None],
+                    "site": [None],
+                    "subpop_pop": [self.population_size],
+                    "subpop_index": [1],
+                    "subpop_name": ["total population"],
+                }
+            )
+        return site_subpop_spine
+
+    @property
+    def date_time_spine(self):
+        if self.data_observed_disease_wastewater is not None:
+            date_time_spine = pl.DataFrame(
+                {
+                    "date": pl.date_range(
+                        start=self.first_wastewater_date,
+                        end=self.last_wastewater_date,
+                        interval="1d",
+                        eager=True,
+                    )
+                }
+            )
+            return date_time_spine.with_columns(
+                pl.arange(0, pl.len()).alias("t")
+            )
+
+    @property
+    def wastewater_data_extended(self):
+        if self.data_observed_disease_wastewater is not None:
+            return (
+                self.data_observed_disease_wastewater.join(
+                    self.date_time_spine, on="date", how="left", coalesce=True
+                )
+                .join(
+                    self.site_subpop_spine,
+                    on=["site_index", "site"],
+                    how="left",
+                    coalesce=True,
+                )
+                .with_columns(
+                    pl.arange(0, pl.len()).alias("ind_rel_to_observed_times")
+                )
+            )
+
+    @property
+    def pop_fraction(self):
+        if self.data_observed_disease_wastewater is not None:
+            subpop_sizes = self.site_subpop_spine.sort(
+                by="subpop_index", descending=False
+            )["subpop_pop"].to_numpy()
```

Comment:
Sorting should not occur within this function, in my opinion. It should return the popsizes in the order that the populations are in the dataframe.